File: 594. Longest Harmonious Subsequence/Python/Solution.py
class Solution:
    def findLHS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        memory = {}
        for num in nums:
            if num not in memory:
                memory[num] = 0
            memory[num] += 1
        result = 0
        for key, value in memory.items():
            if key + 1 in memory:
                result = max(result, value + memory[key + 1])
        return result

File: 594. Longest Harmonious Subsequence/C++/Solution.cpp
class Solution {
public:
    int findLHS(vector<int> &nums) {
        unordered_map<int, int> memory;
        for (int num : nums) {

            memory[num]++;
        }
        int result = 0;
        for (auto item : memory) {
            if (memory.find(item.first + 1) != memory.end()) {
                result = max(result, item.second + memory[item.first + 1]);
            }
        }
        return result;
    }
};